ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -707,44 +711,67 @@ let query_domain_workgroup ~domain =
-      ; Printf.sprintf "realm = %s" domain
-      ; "security = ADS"
-      ; "template shell = /bin/bash"
-      ; "winbind refresh tickets = Yes"
-      ; "winbind enum groups = no"
-      ; "winbind enum users = no"
-      ; Printf.sprintf "winbind cache time = %d" !Xapi_globs.winbind_cache_time
-      ; Printf.sprintf "machine password timeout = %d"
-          !Xapi_globs.winbind_machine_pwd_timeout
-      ; Printf.sprintf "kerberos encryption types = %s"
-          (Kerberos_encryption_types.Winbind.to_string
-             !Xapi_globs.winbind_kerberos_encryption_type
-          )
-      ; Printf.sprintf "workgroup = %s" workgroup
-      ; Printf.sprintf "netbios name = %s" netbios_name
-      ; "idmap config * : range = 3000000-3999999"
-      ; Printf.sprintf "idmap config %s: backend = rid" domain
-      ; Printf.sprintf "idmap config %s: range = 2000000-2999999" domain
-      ; Printf.sprintf "log level = %s" (debug_level ())
-      ; "idmap config * : backend = tdb"
-      ; "" (* Empty line at the end *)
-      ]
+    match domain with
+    | Some dom ->
+        String.concat "\n"
+          [
+            "# auto-generated by xapi"
+          ; "[global]"
+          ; "kerberos method = secrets and keytab"
+          ; Printf.sprintf "realm = %s" dom
+          ; "security = ADS"
+          ; "template shell = /bin/bash"
+          ; "winbind refresh tickets = yes"
+          ; "winbind enum groups = no"
+          ; "winbind enum users = no"
+          ; "winbind scan trusted domains = yes"
+          ; "winbind use krb5 enterprise principals = yes"
+          ; Printf.sprintf "winbind cache time = %d"
+              !Xapi_globs.winbind_cache_time
+          ; Printf.sprintf "machine password timeout = %d"
+              !Xapi_globs.winbind_machine_pwd_timeout
+          ; Printf.sprintf "kerberos encryption types = %s"
+              (Kerberos_encryption_types.Winbind.to_string
+                 !Xapi_globs.winbind_kerberos_encryption_type
+              )
+          ; Printf.sprintf "workgroup = %s" workgroup
+          ; Printf.sprintf "netbios name = %s" netbios_name
+          ; "idmap config * : range = 3000000-3999999"
+          ; Printf.sprintf "idmap config %s: backend = rid" dom
+          ; Printf.sprintf "idmap config %s: range = 2000000-2999999" dom
+          ; Printf.sprintf "log level = %s" (debug_level ())
+          ; Printf.sprintf "allow kerberos auth fallback = %s" allow_fallback
+          ; "idmap config * : backend = tdb"
+          ; "" (* Empty line at the end *)
+          ]
+    | None ->
+        String.concat "\n"
+          [
+            "# autogenerated by xapi"; "[global]"; "" (* Empty line at the end *)
+          ]
   in
+
   let len = String.length conf_contents in
   Unixext.atomic_write_to_file smb_config 0o0644 (fun fd ->
       let (_ : int) = Unix.single_write_substring fd conf_contents 0 len in
       Unix.fsync fd
   )
 
+let clean_winbind_config () =
```

Comment:
Is "clean" here to be read as an adjective or a verb? Using "clear" could be less ambiguous if you want this to be a verb.

Reply:
My dictionary tell me `clean` can also be used as `vert`
but you are expert, I will update.